app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,17 @@
+class SessionsController < ApplicationController
+  def create
+    @user = User.find_or_create_from_auth(auth_hash)
+    self.current_user = @user
+    redirect_to '/'
+  end
+
+  def destroy
+    session[:user_id] = nil
+  end
+
+  protected
+
+  def auth_hash
+    request.env['omniauth.auth']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.